### PR TITLE
Add a tutorial for NIO + UDP

### PR DESCRIPTION
**Pull request by Joannis**

### CODE REVIEW

File: Sources/Articles/Documentation.docc/2024/working-with-udp-in-swiftnio/working-with-udp-in-swiftnio.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+### What is UDP?
+
+UDP is a core networking protocol like TCP, built on top of IPv4 and IPv6. It's a simple protocol that omits delivery checks, which can cause data to be lost. However, this simplicity allows UDP to be faster than TCP, as it doesn't wait for lost packets to be retransmitted.
+
+In this tutorial, you'll learn how to build a UDP server using SwiftNIO. You'll build a simple UDP echo server that listens for incoming packets, reads a string, and sends them back in reverse to the client. By the end of this tutorial, you'll know how to create a UDP server using SwiftNIO and how to send and receive UDP packets.
+
+## How UDP Works
+
+Unlike TCP, UDP sockets don't distinguish between client and server. Any client that sends a packet to a UDP socket needs to bind to a oprt to receive a response. This makes UDP a connectionless protocol, as it doesn't establish a connection before sending data. Instead, it sends packets directly to the destination.
```
Comment (tib) [inline Sources/Articles/Documentation.docc/2024/working-with-udp-in-swiftnio/working-with-udp-in-swiftnio.md:15]: > oprt 

port

File: Sources/Articles/Documentation.docc/2024/working-with-udp-in-swiftnio/working-with-udp-in-swiftnio.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+@Snippet(path: "site/Snippets/working-with-udp-in-swiftnio", slice: "packets")
+
+1. Each packet received in `inbound` is read into a String
+2. The string is reversed, and packet back into a ByteBuffer. This is not very optimised, nor a real use case, but serves a as simple example.
+3. The packet is written back, addressed to the `remoteAddress` that sent the original packet.
+
+### Testing the Server
+
+Now that you've got your code ready, just run the app on your mac. Then, from your terminal, run the following command:
```
Comment (tib) [inline Sources/Articles/Documentation.docc/2024/working-with-udp-in-swiftnio/working-with-udp-in-swiftnio.md:42]: I'd add a sentence about what's nc and what happens here...

**Review by tib — CHANGES_REQUESTED** [review]
Looks good, 2 minor things.

File: Sources/Articles/Documentation.docc/2024/working-with-udp-in-swiftnio/working-with-udp-in-swiftnio.md (hunk 2 above)
Comment (Joannis) [reply Sources/Articles/Documentation.docc/2024/working-with-udp-in-swiftnio/working-with-udp-in-swiftnio.md:42]: That's right under the bash script, do I need something else?

https://github.com/swift-on-server/articles/pull/1/files#diff-67b50b5c7131e794d42ca70e1945bff2b6cd48d1485b7becb97684c2a25e7066R48-R50

File: Sources/Articles/Documentation.docc/2024/working-with-udp-in-swiftnio/working-with-udp-in-swiftnio.md (hunk 1 above)
Comment (Joannis) [inline Sources/Articles/Documentation.docc/2024/working-with-udp-in-swiftnio/working-with-udp-in-swiftnio.md:15]:
```suggestion
Unlike TCP, UDP sockets don't distinguish between client and server. Any client that sends a packet to a UDP socket needs to bind to a port to receive a response. This makes UDP a connectionless protocol, as it doesn't establish a connection before sending data. Instead, it sends packets directly to the destination.
```

File: Sources/Articles/Documentation.docc/2024/working-with-udp-in-swiftnio/working-with-udp-in-swiftnio.md (hunk 2 above)
Comment (tib) [reply Sources/Articles/Documentation.docc/2024/working-with-udp-in-swiftnio/working-with-udp-in-swiftnio.md:42]: I was thinking of something like this: 

The nc (NetCat) command is a popular tool for managing network connections. Its main use is to establish and handle them. After your code is ready, start the app on your Mac, and run the following command in your terminal:

**Review by tib — APPROVED** [review]

**Pull request merged**